### ISSUE TITLE
Tasks for release v1.2 branch and starting v1.3 cycle

### ISSUE THREAD

**Issue by ykakarap** [kubernetes-sigs/cluster-api#6615]
Let's wait with implementation until we get close to the release. No action needed at the moment.

### Setup CI signal for v1.2

* [ ] Create the release branch "release-1.2"
* [ ] Create new jobs and dashboard for the release-1.2 branch in test-infra
    * Previous Issue: #5970
    * Previous PR: https://github.com/kubernetes/test-infra/pull/24983

* [ ] Book:
    * [ ] Make sure the previous book is available under `https://release-1-1.cluster-api.sigs.k8s.io/` and ``https://cluster-api.sigs.k8s.io/``
        * ~ Previous PR: https://github.com/kubernetes/k8s.io/pull/3404
        * Change production branch in Netlify (might need a new rollout or something)
        * Check that certificates are valid / from time to time open a Netlify ticket to update certificate SANs (see https://github.com/kubernetes-sigs/cluster-api/issues/6017)
    * [ ] Update refs in introduction.md (drop unsupported versions, add previous version)
        * ~ https://github.com/kubernetes-sigs/cluster-api/pull/6192
    * [ ] Add a new migration doc for provider implementers
        * Previous PR: https://github.com/kubernetes-sigs/cluster-api/pull/6018
    * [ ] Update supported versions in versions.md
    * [ ] Update clusterctl links in the quickstart
        * Previous PR: https://github.com/kubernetes-sigs/cluster-api/pull/6137

### Starting work on v1.3

* [ ] GitHub milestone
    * [ ] Create the new v1.3 milestone via GitHub UI
    * [ ] Update the [milestone_applier config](https://github.com/kubernetes/test-infra/blob/0b17ef5ffd6c7aa7d8ca1372d837acfb85f7bec6/config/prow/plugins.yaml#L371) accordingly
        * Previous PR: https://github.com/kubernetes/test-infra/pull/24988
    * [ ] Move GitHub issues to the new milestone (schedule triage session @vincepri)

* [ ] Prepare main branch for v1.3 development:
    * Previous Issue: https://github.com/kubernetes-sigs/cluster-api/issues/6020
    * Previous PR: https://github.com/kubernetes-sigs/cluster-api/pull/5982

* [ ] Check if we should remove previously deprecated code
    * Previous Issue: https://github.com/kubernetes-sigs/cluster-api/issues/6103

* [ ] Jobs:
    * [ ] Modify the clusterctl upgrade jobs to test the new release
        * Previous Issue: https://github.com/kubernetes-sigs/cluster-api/issues/6019

* [ ] If we upgrade the apiVersion [DOES NOT APPLY]
    * Add new version of the types:
        * Create new api packages by copying existing packages
        * Make sure webhooks only exist in the latest apiVersion (same for other subpackages like `index`)
        * Add conversion and conversion tests
        * Adjust generate targets in the Makefile
    * Update import aliases in .golangci.yml
    * Switch other code over to the new version (imports across the code base, e.g. controllers)
        * Add all versions to the schema in the `main.go`'s
    * Add types to the `PROJECT` files of the respective provider
    * Add test data for the new version in test/e2e/data/{infrastructure-docker,shared} (also update top-level `.gitignore`)
    * Update docker.yaml, make sure all tests are successful in CI

**Comment by sbueringer** [comment]
@ykakarap Looks good, feel free to ping me if you need reviews somewhere :) (or some discussions around when we can do which step)

**Comment by sbueringer** [comment]
@CecileRobertMichon No rush, but can you write the Netlify folks to fix our cert again? (https://release-1-2.cluster-api.sigs.k8s.io/reference/versions.html)

**Comment by sbueringer** [comment]
@ykakarap I opened a PR for the milestone applier to avoid mislabeling PRs to the wrong milestone

**Comment by sbueringer** [comment]
fyi milestone handling might change for v1.3: https://github.com/kubernetes-sigs/cluster-api/issues/5519#issuecomment-1164820058

**Comment by sbueringer** [comment]
@ykakarap I think we already did ` Update supported versions in versions.md` at some point for v1.2. We could do the same already for v1.3 and move the task down to the v1.3 section

**Comment by sbueringer** [comment]
If I see correctly with have PRs for everything and they are on hold until the release, except:
* Update supported versions in versions.md
* Move GitHub issues to the new milestone: Depends on #6828

**Comment by ykakarap** [comment]
Since we decided to do v1.2.0-beta.2 for the clusterctl upgrade tests for now and go back and do another round of PRs once we have v1.2.0  - those PR should still be considered pending. I will update the task list to update this new information.

**Comment by ykakarap** [comment]
Sounds good. I will create the separate issue, link it here and close this.
